compiler/pragmas.nim

```diff
@@ -968,10 +996,12 @@ proc singlePragma(c: PContext, sym: PSym, n: PNode, i: int,
         else:
           localError(it.info, "'experimental' pragma only valid as toplevel statement")
       of wThis:
-        if it.kind == nkExprColonExpr:
+        if it.kind in nkPragmaCallKinds and it.len == 2:
```

Comment:
I agree with @data-man here. You're using this expression many times.

But what's possibly error prone is that you're using the negation of this too (just above on line 988, and probably in other places too). It would be far clearer if this expression was in some nice proc and you negated it when necessary.

Reply:
That's a bit unfair since the original code had the same issues. I planned to clean this one up after the merge. But ok, since other stuff needs to be reworked too, no big issue.

Reply:
I don't think it was as severe previously: only the kind was checked, no?

Reply:
I disagree. Introduction of a new function for every pair of checks will turn compiler's code into hell, because it will contain 30000 one line helper procs that everybody will have to learn before reading compilers code. 

It is pointless: it will not save a single line of code, because if statements can't be removed as error messages are different for different pragma.

Reply:
I'm going to let @Araq decide. If this was my code though I wouldn't accept it. This is a common check.